### PR TITLE
Bugfix for localization of messages with currencies that haven't cents

### LINKED ISSUE #29405: Localization issue for messages with currencies that have not cents, like KRW
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

There were some critical errors in localization process when currency did not have cents, like KRW.

Unfortunately, we did not keep full details of the bug. But the solution presented in this PR worked fine for more than month already:  https://github.com/PrestaShop/PrestaShop/pull/29404 Please, see details in code.

### Expected behavior

Absence of cents should not affect localization behavior.

### Steps to reproduce

Unfortunately, detailed logging of steps was not performed.

### PrestaShop version(s) where the bug happened

1.7.8.5

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by kamilrafik**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Earlier, there was int type hint in this method; but in the process of developing one e-store with KRW currency that does not have analog of cents and does not need precision, it was found that sometimes null is passed to this method which causes some errors. Detailed testing is needed to reproduce errors, they were not logged fully; but this bugfix with replacing int type hint with intval() call worked fine and did not cause any visible issues. Obviously, the original issue and bugfix should be researched by some experienced PrestaShop core developer.
| Type?             | bug fix
| Category?         | LO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #29405
| Related PRs       | not applicable
| How to test?      | It requires research of the described issue by experienced PrestaShop core developer.
| Possible impacts? | Localization features may be impacted.

### CODE REVIEW

**Review by matks — CHANGES_REQUESTED** [review]
Dear @kamilrafik thank you for this contribution.

I understand you want to help by providing this bugfix however as you mentioned this is working in some usecase but you cannot guarantee it will have no side-effects.

Unless you can provide the detailed testing you mention, we cannot merge this.

The original issue and bugfix will not be researched by some experienced PrestaShop core developer 😅 this is not how open source works. If this bugfix is needed by someone it must be researched by this person.

Your bugfix will be useful for people looking for a workaround 👏 however we cannot merge it as it is not fully explored, understood and testable.

**Comment by kamilrafik** [comment]
This PR was requested by company owner to minimize conflicts on upgrades in the future. If the company will decide to invest money into research of this bug, more details will be posted by me here or in the related issue. If not, it will stay as is.

File: src/Core/Localization/CLDR/ComputingPrecision.php
Comment (yanmakouf) [inline src/Core/Localization/CLDR/ComputingPrecision.php:43]: In order to have a proper fix, it should be investigated why sometimes null is passed. When adding KRW to the currencies the BD table entry  ps_currency.precision is indeed set to 0. So there must be somewhere in the flow where this is corrupted

**Comment by matks** [comment]
> This PR was requested by company owner to minimize conflicts on upgrades in the future. If the company will decide to invest money into research of this bug, more details will be posted by me here or in the related issue. If not, it will stay as is.

Hey @kamilrafik looks like the chosen usecase was "it will stay as is" 😄 

No worries as I mentioned your bugfix will be useful for people looking for a workaround 👏 however we cannot merge it as it is not fully explored, understood and testable.

Since the PR did not moved for 30 days, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale.

Best regards,

**Pull request closed without merging**
